app/src/androidTest/kotlin/com/google/samples/apps/nowinandroid/ui/interests2pane/InterestsListDetailScreenTest.kt

```diff
@@ -0,0 +1,232 @@
+        }
+    }
+
+    @Test
+    fun expandedWidth_topicSelected_updatesDetailPane() {
+        composeTestRule.apply {
+            setContent {
+                with(TestDeviceConfig.Expanded) {
+                    DeviceConfigurationOverride(override = sizeOverride) {
+                        NiaTheme {
+                            InterestsListDetailScreen(windowAdaptiveInfo = adaptiveInfo)
+                        }
+                    }
+                }
+            }
+
+            val firstTopic = runBlocking {
+                topicsRepository.getTopics().first().first()
+            }
+            onNodeWithText(firstTopic.name).performClick()
+
+            onNodeWithTag(listPaneTag).assertIsDisplayed()
+            onNodeWithText(placeholderText).assertIsNotDisplayed()
+            onNodeWithTag(firstTopic.testTag).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun notExpandedWidth_topicSelected_showsTopicDetailPane() {
+        composeTestRule.apply {
+            setContent {
+                with(TestDeviceConfig.Compact) {
+                    DeviceConfigurationOverride(override = sizeOverride) {
+                        NiaTheme {
+                            InterestsListDetailScreen(windowAdaptiveInfo = adaptiveInfo)
+                        }
+                    }
+                }
+            }
+
+            val firstTopic = runBlocking {
+                topicsRepository.getTopics().first().first()
+            }
+            onNodeWithText(firstTopic.name).performClick()
+
+            onNodeWithTag(listPaneTag).assertIsNotDisplayed()
+            onNodeWithText(placeholderText).assertIsNotDisplayed()
+            onNodeWithTag(firstTopic.testTag).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun expandedWidth_backPressFromTopicDetail_leavesInterests() {
+        var unhandledBackPress = false
+        composeTestRule.apply {
+            setContent {
+                with(TestDeviceConfig.Expanded) {
+                    DeviceConfigurationOverride(override = sizeOverride) {
+                        NiaTheme {
+                            // Back press should not be handled by the two pane layout, and thus
+                            // "fall through" to this BackHandler.
+                            BackHandler {
+                                unhandledBackPress = true
+                            }
+                            InterestsListDetailScreen(windowAdaptiveInfo = adaptiveInfo)
+                        }
+                    }
+                }
+            }
+
+            val firstTopic = runBlocking {
+                topicsRepository.getTopics().first().first()
+            }
+            onNodeWithText(firstTopic.name).performClick()
+
+            Espresso.pressBack()
+
+            assertTrue(unhandledBackPress)
+        }
+    }
+
+    @Test
+    fun notExpandedWidth_backPressFromTopicDetail_showsListPane() {
```

Comment:
Nit: I think all `notExpandedWidth` be `compactWidth`, which is more descriptive, since `notExpandedWidth` could also mean `mediumWidth`
```suggestion
    fun compactWidth_backPressFromTopicDetail_showsListPane() {
```

Reply:
Agreed, fixed